From 759804b5a067c70889dd703789bc470e9bb72861 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Tue, 24 Nov 2015 00:01:33 +0800
Subject: [PATCH 0555/1240] fix: clock: a3700: update counter frequency

issue: previous Armada-3700 counter frequency
      is not accurate, which leads to less delay
      for many drivers.
fix: correct the counter frequency from 25MHZ
      to 80MHZ.
      Current system counter clock source is derived
      from TBG-BS, then it will be divided by
      COUNTER_CLK_PRSCL and CPU_CNT_CLK_DIV_SEL.
      Current value, TBG-BS: 800MHZ, CPU_CNT_CLK_DIV_SEL:
      divide by 2, COUNTER_CLK_PRSCL: divide by 5,
      result: 800/2/5 = 80MHZ.
      Please PAY ATTENTION that this frequency should be
      changed once the TBG-BS is changed.

Change-Id: Iec65b880c763ed05d403d6237381bb98478c8165
Reviewed-on: http://vgitil04.il.marvell.com:8080/25183
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/25458
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 include/configs/armadalp.h | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/include/configs/armadalp.h b/include/configs/armadalp.h
index 1398982..54bf355 100644
--- a/include/configs/armadalp.h
+++ b/include/configs/armadalp.h
@@ -48,8 +48,21 @@
 /* Plaform */
 #define CONFIG_MARVELL
 
+/* Clock Definitions */
+#ifdef CONFIG_PALLADIUM
 #define COUNTER_FREQUENCY	(6000)
-
+#else
+/*
+ * Current system counter clock source is derived from TBG-BS,
+ * then it will be divided by COUNTER_CLK_PRSCL and CPU_CNT_CLK_DIV_SEL.
+ * Current value, TBG-BS: 800MHZ, CPU_CNT_CLK_DIV_SEL: divide by 2,
+ * COUNTER_CLK_PRSCL: divide by 5, result: 800/2/5 = 80MHZ
+ * Please PAY ATTENTION this frequency should be changed accordingly
+ * if counter clock source, TBG-BS, COUNTER_CLK_PRSCL or CPU_CNT_CLK_DIV_SEL
+ * are changed.
+ */
+#define COUNTER_FREQUENCY	(80 * 1000000)
+#endif
 
 /* DRAM detection stuff */
 #define MV_DRAM_AUTO_SIZE
-- 
1.9.1

